lib/removeUnusedMaterials.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removeUnusedMaterials;
+
+function removeUnusedMaterials(gltf, stats) {
+    var usedMaterialIds = {};
+    var meshes = gltf.meshes;
+
+    // Build hash of used materials by iterating through meshes
+    if (defined(meshes)) {
+        for (var meshId in meshes) {
+            if (meshes.hasOwnProperty(meshId)) {
+                if (defined(meshes[meshId].primitives)) {
+                    var length = meshes[meshId].primitives.length;
```

Comment:
Instead of repeating `meshes[meshId].primitives` several times here, it is cleaner and faster to store it in a local, e.g.,
```
var primitives = meshes[meshId].primitives;
```